core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/LandingZoneCreationStatusRunner.scala

```diff
@@ -0,0 +1,127 @@
+          case JobReport.StatusEnum.FAILED => "Failure Reported, but no errors returned"
+          case JobReport.StatusEnum.SUCCEEDED if result.getLandingZone == null =>
+            "Result marked as success, but no landing zone returned"
+          case JobReport.StatusEnum.RUNNING | JobReport.StatusEnum.SUCCEEDED =>
+            "Invalid Failure" // FIXME: improve message (we should never get into this state)
+          case null => "No status reported"
+        }
+      // no job report, no error report (or at least no error report message), also no lz (since that would still be a success)
+      case _ => "No landing zone, job report, or errors returned from landing zone creation"
+    }
+    Some(s"Landing Zone creation failed: $msg")
+  }
+
+  override def run(
+    job: WorkspaceManagerResourceMonitorRecord
+  )(implicit executionContext: ExecutionContext): Future[JobStatus] = {
+    val billingProjectName = job.billingProjectId match {
+      case Some(name) => RawlsBillingProjectName(name)
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneResult created with id ${job.jobControlId} but no billing project set"
+        )
+        return Future.successful(Complete) // nothing more this runner can do with it
+    }
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneResult for billing project $billingProjectName created with id ${job.jobControlId} but no user email set"
+        )
+        val errorMsg =
+          s"Unable to update ${billingProjectName.value} with landing zone status because no user email is stored on monitoring job"
+        billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, Some(errorMsg))
+        return Future.successful(Complete)
+    }
+    getUserCtx(userEmail)
+      .map(userCtx =>
+        Try(workspaceManagerDAO.getCreateAzureLandingZoneResult(job.jobControlId.toString, userCtx)) match {
+          case Failure(exception) =>
+            val message = Some(s"Api call to get landing zone from workspace manager failed: ${exception.getMessage}")
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, message)
+            Incomplete
+          case Success(result) if result.getJobReport != null =>
+            result.getJobReport.getStatus match {
+              // the job just isn't done yet - return indicates this runner isn't finished with the job so it's rescheduled
+              case JobReport.StatusEnum.RUNNING => Incomplete
+              // a normal success
+              case JobReport.StatusEnum.SUCCEEDED if result.getLandingZone != null =>
+                billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Ready, None)
+                billingRepository.updateLandingZoneId(billingProjectName, result.getLandingZone.getId)
```

Comment:
I don't think you need to worry about updating the landing zone ID-- we set it at creation time and there is no reason it should change.